src/coreclr/gc/gcinterface.h

```diff
@@ -981,6 +1001,12 @@ class IGCHeap {
 
     // Refresh the memory limit
     virtual int RefreshMemoryLimit() PURE_VIRTUAL
+
+    // Enable NoGCRegionCallback
+    virtual enable_no_gc_region_callback_status EnableNoGCRegionCallback(NoGCRegionCallbackFinalizerWorkItem* callback, uint64_t callback_threshold) PURE_VIRTUAL
+
+    // Get extra work for the finalizer
+    virtual FinalizerWorkItem* GetFinalizerWork() PURE_VIRTUAL
```

Comment:
>GetFinalizerWork [](http://example.com/codeflow?start=31&length=16)

this name seems very generic and to be only for calling user provided callbacks while the majority of fianalization work is not using this at all. how about we name it something like `GetFinalizerUserCallback`?

Reply:
or `GetUserWorkForFinalization`

Reply:
This mechanism can be used to schedule any kind of work for execution on the finalizer thread. Do you believe that we will only ever need it for "user" work?

Reply:
How about `GetExtraWorkForFinalization`?

Reply:
the majority of the work scheduled on the finalizer thread by the GC is running finalizers, and that's not done through this API. I'm fine with Extra work too. just to make it clear that this is not the usual work that GC schedules on the finalizer thread.